manifests/server/etcd.pp

```diff
@@ -1,12 +1,31 @@
 # @summary Sets up an etcd cluster node
+#
+# @param addn_names additional names for certificates
+# @param cert_path path to cert files
+# @param client_ca_cert
+# @param client_ca_key
+# @param cluster_name name of the etcd cluster for searching its nodes in the puppetdb
+# @param ensure set ensure for installation or deinstallation
+# @param firewall_type define the type of firewall to use
+# @param generate_ca whether to generate a own ca or not
+# @param manage_certs whether to manage certs or not
+# @param manage_firewall whether to manage firewall or not
+# @param manage_members whether to manage the ectd cluster member joining or not
+# @param manage_setup whether to manage the setup of etcd or not
+# @param peer_ca_cert
+# @param peer_ca_key
+# @param puppetdb_discovery_tag enable puppetdb resource searching
+# @param self_signed_tls
+# @param version version of ectd to install
+#
 class k8s::server::etcd (
   K8s::Ensure $ensure = 'present',
   String[1] $version  = pick($k8s::etcd_version, '3.5.1'),
 
   Boolean $manage_setup             = true,
   Boolean $manage_firewall          = false,
   Boolean $manage_members           = false,
-  String[1] $cluster_name           = 'default',
+  String[1] $cluster_name           = pick($k8s::server::etcd_cluster_name, 'default'),
```

Comment:
why is the pick required, since `default` is defined as default value?

Reply:
@ananace said in another comment:

> This should be a `pick($k8s::server::etcd_cluster_name, 'default')`, the k8s classes might not be included on nodes that are purely etcd servers, so having a fallback on it is necessary.